gateway_tests/conftest.py

```diff
@@ -434,3 +434,116 @@ def compare_structures(gw_struct, ioc_struct) -> str:
                 f"Element '{key}' : GW has '{gateway_value}', IOC has '{ioc_value}'"
             )
     return "\n\t".join(differences)
+
+
+@contextlib.contextmanager
+def ca_subscription(
+    pvname: str,
+    callback: PyepicsCallback,
+    mask: int = epics.dbr.DBE_VALUE,
+    form: str = "time",
+    count: int = 0,
+    timeout: float = 0.5,
+) -> ContextManager[int]:
+    """Low-level ``epics.ca`` subscription. Yields channel identifier."""
+    event_id = None
+    chid = epics.ca.create_channel(pvname)
+    try:
+        connected = epics.ca.connect_channel(chid, timeout=timeout)
+        assert connected, f"Could not connect to channel: {pvname}"
+
+        (_, _, event_id) = epics.ca.create_subscription(
+            chid,
+            mask=mask,
+            use_time=form == "time",
+            use_ctrl=form == "ctrl",
+            callback=callback,
+            count=count,
+        )
+        yield chid
+    finally:
+        if event_id is not None:
```

Comment:
I'm not sure if this is ever set to `None` by pyepics, it's not clear from the EPICS docs + the pyepics source/docs that this can be anything other than a ctypes pointer
https://github.com/pyepics/pyepics/blob/418858d20790e8c76fd7e53f577bc52e2769bb87/epics/ca.py#L1881-L1888
And EPICS says the last argument to `ca_create_subscription` is:
```
PEVID
This is a pointer to user supplied event id which is overwritten if successful. This event id can later be used to clear a specific event. This option may be omitted by passing a null pointer.
```
Am I misreading here?

Reply:
Nope, you've got it right as to `ca_create_subscription`. The `None` check here is rather for the `try/finally` block (create channel but not subscription? please clean up the channel in `finally`).

Reply:
Ahhh

Reply:
Ok cool marking resolved